### PR TITLE
Automated pre release workflow

### PR DESCRIPTION
**Pull request by pdhanoyasplunk**
Added Pre Release workflow : 
- Creates a PR with all the changes needed for release. 
- Tags and push RC images
- Generates artifacts to be distributed to QA/Test Teams with RC image

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1 +1,2 @@
 OPERATOR_SDK_VERSION=v0.18.2
+REVIEWERS=pdhanoyasplunk,smohan-splunk,kashok-splunk,akondur,sgontla
```
Comment (sgontla) [inline .env:2]: Not sure how this works, if it needs, then we can add all the team?

**Review by sgontla — APPROVED** [review]

**Review by kashok-splunk — APPROVED** [review]

**Review by smohan-splunk — APPROVED** [review]

**Review by akondur — COMMENTED** [review]
Great work @pdhanoyasplunk. A few minor comments. A sample example PR would be great with the final code changes.

File: .github/pre-release-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+    - name: Update Operator Version in version.go
+      if: github.event.inputs.old_operator_version != github.event.inputs.new_operator_version
+      uses: jacobtomlinson/gha-find-replace@master
+      with:
+        find: "${{ github.event.inputs.old_operator_version }}"
+        replace: "${{ github.event.inputs.new_operator_version }}"
+        include: "version.go"
+
+    - name: Update Operator Image name in DOCS
```
[the remark this answers is not in the document]
Comment (pdhanoyasplunk) [reply .github/pre-release-workflow.yml:92]: I did update the code to `${{ github.event.inputs.old_enterprise_version }} or later"` from `(${{ github.event.inputs.old_enterprise_version }} or later")` that should take care of the extra braces.

File: build/make_bundle.sh [hunk 3]
```diff
@@ -134,8 +137,7 @@ cat << EOF >$YAML_SCRIPT_FILE
 EOF
 
 echo Updating $OLM_CATALOG
-operator-sdk generate csv --csv-version $VERSION --operator-name splunk --update-crds --make-manifests=false --verbose
-yq w -i -s $YAML_SCRIPT_FILE $OLM_CATALOG/splunk/$VERSION/splunk.v${VERSION}.clusterserviceversion.yaml
```
[the remark this answers is not in the document]
Comment (pdhanoyasplunk) [reply build/make_bundle.sh:138]: I just updated the command to include `--from-version $LAST_RELEASE_VERSION` the diff is showing it as deleted.

File: .env (hunk 1 above)
Comment (pdhanoyasplunk) [reply .env:2]: Yeah we can add whole team as well will need to include the all the usernames. I can do that if it's needed.

File: .github/pre-release-workflow.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (pdhanoyasplunk) [reply .github/pre-release-workflow.yml:92]: Here is an updated PR: https://github.com/pdhanoyasplunk/splunk-operator/pull/18/files

File: build/make_bundle.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (akondur) [reply build/make_bundle.sh:138]: There is a line `yq w -i -s $YAML_SCRIPT_FILE $OLM_CATALOG/splunk/$VERSION/splunk.v${VERSION}.clusterserviceversion.yaml` which is deleted?

**Review by akondur — APPROVED** [review]

**Review by smohan-splunk — APPROVED** [review]

**Pull request merged**